### PR TITLE
🎉  Source Intercom: Update Intercom API to version 2.5

### LINKED ISSUE #15141: Source Intercom: Add custom_attributs in conversation schema
### Tell us about the problem you're trying to solve
I'd like to retrieve custom_attributes from intercom conversations while using intercom source. 

### Describe the solution you’d like
Add custom_attributes in intercon conversation schema.

### Describe the alternative you’ve considered or used
None.

### Additional context
None.

### Are you willing to submit a PR?
Yes.

### PR DESCRIPTION
**Pull request by alex-gron**
## What
Upgrade Intercom API from v2.2 to v2.5
Primary use case is to pull in `custom_attributes` on the Conversation object
Closes #15141

## How
Increment API version to 2.5
Update schema.json to include `custom_attributes`

## 🚨 User Impact 🚨
Backwards compatible with earlier API versions

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [x] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

See below

</details>

<details><summary><strong>Integration</strong></summary>

See below

</details>

<details><summary><strong>Acceptance</strong></summary>

See below

</details>

### CODE REVIEW

**Comment by alex-gron** [comment]
Unit Tests:

```
(.venv) alexgronemeyer@Alexs-MacBook-Pro source-intercom % python -m pytest unit_tests                                       
Test session starts (platform: darwin, Python 3.9.11, pytest 6.1.2, pytest-sugar 0.9.5)
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/alexgronemeyer/src/airbyte/airbyte-integrations/connectors/source-intercom/.hypothesis/examples')
rootdir: /Users/alexgronemeyer/src/airbyte, configfile: pytest.ini
plugins: sugar-0.9.5, requests-mock-1.9.3, mock-3.6.1, hypothesis-6.54.3, timeout-1.4.2, cov-3.0.0
collecting ... 
 airbyte-integrations/connectors/source-intercom/unit_tests/test_cached_stream_state.py::test_full_refresh[Sync Started. Parent.] ✓                             3% ▍         
 airbyte-integrations/connectors/source-intercom/unit_tests/test_cached_stream_state.py::test_full_refresh[Sync Started. Child.] ✓                              6% ▋         
 airbyte-integrations/connectors/source-intercom/unit_tests/test_cached_stream_state.py::test_incremental_sync[Sync Started. Parent] ✓                          9% ▉         
 airbyte-integrations/connectors/source-intercom/unit_tests/test_cached_stream_state.py::test_incremental_sync[Sync Started. Child] ✓                          12% █▍        
 airbyte-integrations/connectors/source-intercom/unit_tests/test_cached_stream_state.py::test_incremental_sync[Sync in progress. Parent] ✓                     16% █▋        
 airbyte-integrations/connectors/source-intercom/unit_tests/test_cached_stream_state.py::test_incremental_sync[Sync in progress. Child] ✓                      19% █▉        
 airbyte-integrations/connectors/source-intercom/unit_tests/test_control_rate_limit.py::test_with_unknown_load[On UNKNOWN load] ✓                              22% ██▎       
 airbyte-integrations/connectors/source-intercom/unit_tests/test_control_rate_limit.py::test_with_unknown_load[On Low Load] ✓                                  25% ██▌       
 airbyte-integrations/connectors/source-intercom/unit_tests/test_control_rate_limit.py::test_with_unknown_load[On Mid Load] ✓                                  28% ██▊       
 airbyte-integrations/connectors/source-intercom/unit_tests/test_control_rate_limit.py::test_with_unknown_load[On High Load] ✓                                 31% ███▎      
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_get_next_page_token[base pagination] ✓                                          34% ███▌      
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_get_next_page_token[companies pagination] ✓                                     38% ███▊      
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_get_next_page_token[contacts pagination] ✓                                      41% ████▏     {"type": "LOG", "log": {"level": "INFO", "message": "Backing off _send(...) for 0.0s (airbyte_cdk.sources.streams.http.exceptions.UserDefinedBackoffException: Request URL: https://api.intercom.io/companies/scroll?per_page=50&scroll_param=expired_scroll_param, Response Code: 404, Response Text: {\"errors\": [{\"code\": \"not_found\", \"message\": \"scroll parameter not found\"}]})"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Retrying. Sleeping for 0.01 seconds"}}
{"type": "LOG", "log": {"level": "ERROR", "message": "Can't create a new scroll request within an minute or scroll param was expired. Let's try to use a standard non-scroll endpoint."}}

 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_switch_to_standard_endpoint_if_scroll_expired ✓                                 44% ████▍     
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_check_connection_ok ✓                                                           47% ████▊     
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_check_connection_empty_config ✓                                                 50% █████     
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_check_connection_invalid_config ✓                                               53% █████▍    
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_check_connection_exception ✓                                                    56% █████▋    {"type": "LOG", "log": {"level": "INFO", "message": "Using start_date: 1647752400.0"}}

 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_streams ✓                                                                       59% █████▉    
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_read[Admins-GET-/admins-response0-expected0] ✓                                  62% ██████▍   
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_read[Companies-GET-/companies/scroll-response1-expected1] ✓                     66% ██████▋   
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_read[CompanySegments-GET-/companies/id/segments-response2-expected2] ✓          69% ██████▉   
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_read[Contacts-POST-/contacts/search-response3-expected3] ✓                      72% ███████▎  
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_read[Conversations-POST-/conversations/search-response4-expected4] ✓            75% ███████▌  
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_read[ConversationParts-GET-/conversations/id-response5-expected5] ✓             78% ███████▊  
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_read[CompanyAttributes-GET-/data_attributes-response6-expected6] ✓              81% ████████▎ 
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_read[ContactAttributes-GET-/data_attributes-response7-expected7] ✓              84% ████████▌ 
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_read[Segments-GET-/segments-response8-expected8] ✓                              88% ████████▊ 
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_read[Tags-GET-/tags-response9-expected9] ✓                                      91% █████████▏
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_read[Teams-GET-/teams-response10-expected10] ✓                                  94% █████████▍
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_conversation_part_has_conversation_id ✓                                         97% █████████▊
 airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_conversation_part_filtering_based_on_conversation ✓                            100% ██████████
============================================================================= warnings summary ==============================================================================
unit_tests/test_cached_stream_state.py:12
  /Users/alexgronemeyer/src/airbyte/airbyte-integrations/connectors/source-intercom/unit_tests/test_cached_stream_state.py:12: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    INTERCOM_STREAM = Conversations(authenticator=NoAuth(), start_date=0)

.venv/lib/python3.9/site-packages/deprecated/classic.py:173: 4 warnings
airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py: 54 warnings
  /Users/alexgronemeyer/src/airbyte/airbyte-integrations/connectors/source-intercom/.venv/lib/python3.9/site-packages/deprecated/classic.py:173: DeprecationWarning: Call to deprecated class HttpAuthenticator. (Use requests.auth.AuthBase instead) -- Deprecated since version 0.1.20.
    return old_new1(cls, *args, **kwargs)

.venv/lib/python3.9/site-packages/airbyte_cdk/sources/streams/http/http.py:41: 2 warnings
airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py: 40 warnings
  /Users/alexgronemeyer/src/airbyte/airbyte-integrations/connectors/source-intercom/.venv/lib/python3.9/site-packages/airbyte_cdk/sources/streams/http/http.py:41: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    self._authenticator: HttpAuthenticator = NoAuth()

unit_tests/test_cached_stream_state.py:13
  /Users/alexgronemeyer/src/airbyte/airbyte-integrations/connectors/source-intercom/unit_tests/test_cached_stream_state.py:13: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    INTERCOM_SUB_STREAM = ConversationParts(authenticator=NoAuth(), start_date=0)

unit_tests/unit_test.py:27
  /Users/alexgronemeyer/src/airbyte/airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py:27: DeprecationWarning: Call to deprecated class AirbyteLogger. (Use logging.getLogger('airbyte') instead) -- Deprecated since version 0.1.47.
    logger = AirbyteLogger()

airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_switch_to_standard_endpoint_if_scroll_expired
  /Users/alexgronemeyer/src/airbyte/airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py:87: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    stream1 = Companies(authenticator=NoAuth())

airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_streams
  /Users/alexgronemeyer/src/airbyte/airbyte-integrations/connectors/source-intercom/source_intercom/source.py:526: DeprecationWarning: Call to deprecated class AirbyteLogger. (Use logging.getLogger('airbyte') instead) -- Deprecated since version 0.1.47.
    AirbyteLogger().log("INFO", f"Using start_date: {config['start_date']}")

airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py: 11 warnings
  /Users/alexgronemeyer/src/airbyte/airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py:210: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    stream = stream(authenticator=NoAuth(), start_date=0)

airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_conversation_part_has_conversation_id
  /Users/alexgronemeyer/src/airbyte/airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py:309: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    conversation_parts = ConversationParts(authenticator=NoAuth())

airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py::test_conversation_part_filtering_based_on_conversation
  /Users/alexgronemeyer/src/airbyte/airbyte-integrations/connectors/source-intercom/unit_tests/unit_test.py:335: DeprecationWarning: Call to deprecated class NoAuth. (Set `authenticator=None` instead) -- Deprecated since version 0.1.20.
    conversation_parts = ConversationParts(authenticator=NoAuth(), start_date=0)

-- Docs: https://docs.pytest.org/en/stable/warnings.html

Results (21.03s):
      32 passed
```

File: airbyte-integrations/connectors/source-intercom/source_intercom/schemas/conversations.json
Comment (ChristopheDuong) [inline airbyte-integrations/connectors/source-intercom/source_intercom/schemas/conversations.json:123]: Would we want to optionally specify in the connector spec (configuration) the custom JSON Schema for that field though? If left empty, it could default to object.
 
But if the user is able to describe what kind of data is being recorded there, then normalization would be able to extra more structured data out...

Is that worth exposing to the connector configuration?

Comment (alex-gron) [reply airbyte-integrations/connectors/source-intercom/source_intercom/schemas/conversations.json:123]: The ultimate goal is to bring in in all of the `custom_attributes` that are configured in Intercom (similar to how the Salesforce connector works to bring in all custom fields). 

The data is stored as "key:value" in Intercom ([source](https://developers.intercom.com/intercom-api-reference/reference/conversation-model#:~:text=will%20return%20null.-,custom_attributes,different%20custom%20attributes%20associated%20to%20the%20conversation%20as%20key%2Dvalue%20pairs.,-open)), so I don't think we need to allow for JSON in the config since it is relatively simple.

Let me know if I am misunderstanding here though!

Comment (ChristopheDuong) [reply airbyte-integrations/connectors/source-intercom/source_intercom/schemas/conversations.json:123]: Is it possible for you to share an example of a record from intercom that includes data in the "custom_attributes" column?

Here in the test / expected results from this PR, it's left empty:

```
"custom_attributes": {},
```

So it's not super insightful on how it'd be presented and used by users afterwards

Comment (alex-gron) [reply airbyte-integrations/connectors/source-intercom/source_intercom/schemas/conversations.json:123]: Yes - we don't have any custom attributes created on our Test Intercom account, so that is why they are empty in the example and integration test.

Here is an example result from the prod Intercom account: 
```
"custom_attributes":{"Needed Escalation":false,"Follow Up Required":false,"Chat Closed Reason":"Alpha Connector Issue"}
```

Comment (alex-gron) [reply airbyte-integrations/connectors/source-intercom/source_intercom/schemas/conversations.json:123]: It ends up in the database as a json column where you can extract the keys/values as follows:
```
select 
    json_extract_scalar(custom_attributes, "$.Needed Escalation") as needed_escalation,
    json_extract_scalar(custom_attributes, "$.Follow Up Required") as follow_up_required,
    json_extract_scalar(custom_attributes, "$.Chat Closed Reason") as chat_closed_reason
from conversations
```

**Review by ChristopheDuong — APPROVED** [review]
I'm not blocking this PR but I had a question on the json schema for the new field (could be re-worked later too but we'd have to make sure to file an issue for it)

Before merging, it still need to publish and update changelog of the connector btw

File: airbyte-integrations/connectors/source-intercom/source_intercom/schemas/conversations.json
Comment (ChristopheDuong) [reply airbyte-integrations/connectors/source-intercom/source_intercom/schemas/conversations.json:123]: yes, so when running the `discover` command, if the catalog would be able to contain:

```
"custom_attributes": {
      "type": ["null", "object"]
      "properties": {
            "Needed Escalation": {
                    "type": ["null", "string"]
            },
            "Follow Up Required": {
                    "type": ["null", "string"]
            },
            "Chat Closed Reason": {
                    "type": ["null", "string"]
            }
      } 
}
```

then normalization would be able to generate the json_extract lines that you linked automatically too

Comment (bazarnov) [reply airbyte-integrations/connectors/source-intercom/source_intercom/schemas/conversations.json:123]: The logic for custom_attributes should be applied, we need to nest all keys into "key" and all the value into "value, than modify the json_schema with this change and it will work correctly for all possible key:value pairs custom_attributes holds.

We can do this now or later on, if we move with the change like in this PR - it's totally ok, but there is a right way to do it, so probably we will make it in the future updates.

**Review by bazarnov — APPROVED** [review]
LGTM, but the change log should be updated, make sure you've done that.

**Pull request merged**
